java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java

```diff
@@ -0,0 +1,410 @@
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.complex;
+
+import static org.apache.arrow.memory.util.LargeMemoryUtil.capAtMaxInt;
+
+import java.util.Collections;
+import java.util.Iterator;
+
+import org.apache.arrow.memory.ArrowBuf;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.util.CommonUtil;
+import org.apache.arrow.util.Preconditions;
+import org.apache.arrow.vector.AddOrGetResult;
+import org.apache.arrow.vector.BaseFixedWidthVector;
+import org.apache.arrow.vector.BaseValueVector;
+import org.apache.arrow.vector.BaseVariableWidthVector;
+import org.apache.arrow.vector.DensityAwareVector;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.NullVector;
+import org.apache.arrow.vector.UInt4Vector;
+import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.ZeroVector;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.util.CallBack;
+import org.apache.arrow.vector.util.OversizedAllocationException;
+import org.apache.arrow.vector.util.SchemaChangeRuntimeException;
+
+public abstract class BaseRepeatedValueViewVector extends BaseValueVector
+    implements RepeatedValueVector, BaseListVector {
+
+  public static final FieldVector DEFAULT_DATA_VECTOR = ZeroVector.INSTANCE;
+  public static final String DATA_VECTOR_NAME = "$data$";
+
+  public static final byte OFFSET_WIDTH = 4;
+  public static final byte SIZE_WIDTH = 4;
+  protected ArrowBuf offsetBuffer;
+  protected ArrowBuf sizeBuffer;
+  protected FieldVector vector;
+  protected final CallBack repeatedCallBack;
+  protected int valueCount;
+  protected long offsetAllocationSizeInBytes = INITIAL_VALUE_ALLOCATION * OFFSET_WIDTH;
+  protected long sizeAllocationSizeInBytes = INITIAL_VALUE_ALLOCATION * SIZE_WIDTH;
+  private final String name;
+
+  protected String defaultDataVectorName = DATA_VECTOR_NAME;
+
+  protected BaseRepeatedValueViewVector(String name, BufferAllocator allocator, CallBack callBack) {
+    this(name, allocator, DEFAULT_DATA_VECTOR, callBack);
+  }
+
+  protected BaseRepeatedValueViewVector(
+      String name, BufferAllocator allocator, FieldVector vector, CallBack callBack) {
+    super(allocator);
+    this.name = name;
+    this.offsetBuffer = allocator.getEmpty();
+    this.sizeBuffer = allocator.getEmpty();
+    this.vector = Preconditions.checkNotNull(vector, "data vector cannot be null");
+    this.repeatedCallBack = callBack;
+    this.valueCount = 0;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public boolean allocateNewSafe() {
+    boolean dataAlloc = false;
+    try {
+      allocateBuffers();
+      dataAlloc = vector.allocateNewSafe();
+    } catch (Exception e) {
+      clear();
+      return false;
+    } finally {
+      if (!dataAlloc) {
+        clear();
+      }
+    }
+    return dataAlloc;
+  }
```

Comment:
I see BaseRepeatedValueVector has the same problem, but shouldn't we free the existing offset/length buffers?

Reply:
I think what's happening is, the sub-class `ListViewvector` calls its `clear` method. And within that method we have a `super.clear()` call. Do you think this is not neat?

Reply:
it makes me question more strongly why these are two separate classes!

Reply:
Yes, I get your point. It shouldn't be like that. Let me see if we can do that clear in the super class itself and avoid calling that in the child, but the issue could be the following. 

The super class keeps the `sizeBuffer` and `offsetBuffer` while the child class keeps the `validityBuffer`. This is what is being done in the `ListVector` and I just used the same pattern. It makes the `clear` logic and rest of the `allocation` logic being done in two classes. 

Do you have a preferred approach?

Reply:
Just note the discrepancy so that it's clear to future readers. In a follow up we may want to merge the classes as I think there's no good reason to have two different classes, but I recognize that it's just following the existing pattern.

Reply:
For the moment I added a comment in the `clear` function call. But I also added a `TODO` not in the top of the class stating the necessity for merging the superclass into the sub-class. 

Thanks for noting this. I will work on this merging separately.